src/Tmds.Ssh/SftpChannel.cs

```diff
@@ -1052,7 +1052,13 @@ private async ValueTask DownloadLinkAsync(string remotePath, string localPath, b
     public ValueTask DownloadFileAsync(string remoteFilePath, string localFilePath, bool overwrite = false, CancellationToken cancellationToken = default)
         => DownloadFileAsync(remoteFilePath, localFilePath, length: null, overwrite, permissions: null, throwIfNotFound: true, cancellationToken);
 
-    private async ValueTask DownloadFileAsync(string remotePath, string localPath, long? length, bool overwrite, UnixFilePermissions? permissions, bool throwIfNotFound, CancellationToken cancellationToken)
+    public ValueTask DownloadFileAsync(string remotePath, string localFilePath, long? length, bool overwrite, UnixFilePermissions? permissions, bool throwIfNotFound, CancellationToken cancellationToken)
+        => DownloadFileAsync(remotePath, (object)localFilePath, length, overwrite, permissions, throwIfNotFound, cancellationToken);
+
+    public ValueTask DownloadFileAsync(string remotePath, Stream destination, long? length, UnixFilePermissions? permissions, bool throwIfNotFound, CancellationToken cancellationToken)
+        => DownloadFileAsync(remotePath, destination, length, false, permissions, throwIfNotFound, cancellationToken);
+
+    private async ValueTask DownloadFileAsync(string remotePath, object destination, long? length, bool overwrite, UnixFilePermissions? permissions, bool throwIfNotFound, CancellationToken cancellationToken)
```

Comment:
Replace the `object destination` arg by separate `string? localFilePath, Stream? destination` args.

Reply:
will do, so you like 2 params instead of 1 :)

Reply:
It works well with https://github.com/tmds/Tmds.Ssh/pull/266/files/72c23ea76d683e464dee46cd08b191af896f57f4#diff-e75f1a3245a183befaa84212ba7019b20d19d6ea2425d893f2e5cf8d5ec84f5f because we'll end up using the `Stream` argument in both cases.